### PR TITLE
Some i128 tests

### PR DESCRIPTION
**Pull request by est31**
* Add some FFI tests for i128 on architectures where we have sort of working "C" FFI support. On all other architectures we ignore the test.
* enhance the u128 overflow tests

### CODE REVIEW

**Comment by est31** [comment]
The makefile looks a bit contrived, but its the only way (the compiletest tool doesn't set stage0 or "windows" cfg's).

File: src/test/run-make/i128-ffi/test.c [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
```
Comment (nagisa) [inline src/test/run-make/i128-ffi/test.c]: Just use the rust_test_helpers.c file that we have precisely for the C FFI testing.

Comment (nagisa) [inline src/test/run-make/i128-ffi/test.c]: This test then can become a plain run-pass one.

**Comment by est31** [comment]
Okay, no more makefiles, thanks to @nagisa

**Comment by nagisa** [comment]
The error seems to be relevant. `(1i128 << 127) == i128::min_value()` and negating min_value of anything will always overflow.

**Comment by est31** [comment]
Okay, I think I've fixed it now. Also, the 128 ffi test got a comment about why its ignored on windows.

r? @eddyb

**Pull request merged**